src/transxchange/read.rs

```diff
@@ -141,24 +142,54 @@ fn read_transxchange(transxchange: &Element, collections: &mut Collections) -> R
     unimplemented!()
 }
 
-fn read_transxchange_archive<P>(transxchange_path: P, collections: &mut Collections) -> Result<()>
+fn read_transxchange_from_zip<P>(transxchange_path: P, collections: &mut Collections) -> Result<()>
 where
     P: AsRef<Path>,
 {
     let zip_file = File::open(transxchange_path)?;
     let mut zip_archive = ZipArchive::new(zip_file)?;
     for index in 0..zip_archive.len() {
-        let mut zip_file = zip_archive.by_index(index)?;
-        match zip_file.sanitized_name().extension() {
+        let mut file = zip_archive.by_index(index)?;
+        let file_name = file.sanitized_name();
+        let file_extension = file_name.extension();
+        match file_extension {
             Some(ext) if ext == "xml" => {
-                info!("reading TransXChange file {:?}", zip_file.sanitized_name());
+                info!("reading TransXChange file {:?}", file_name);
                 let mut file_content = String::new();
-                zip_file.read_to_string(&mut file_content)?;
+                file.read_to_string(&mut file_content)?;
                 let root: Element = file_content.parse()?;
                 read_transxchange(&root, collections)?;
             }
             _ => {
-                info!("skipping file in zip: {:?}", zip_file.sanitized_name());
+                info!("skipping file in zip: {:?}", file_name);
+            }
+        }
+    }
+    Ok(())
+}
+
+fn read_transxchange_from_path<P>(transxchange_path: P, collections: &mut Collections) -> Result<()>
+where
+    P: AsRef<Path>,
+{
+    for entry in WalkDir::new(transxchange_path)
+        .into_iter()
+        .filter_map(|e| e.ok())
```

Comment:
what happens if it is not ok?

Reply:
> Or, if you'd like to iterate over all entries and ignore any errors that may arise, use filter_map. (e.g., This code below will silently skip directories that the owner of the running process does not have permission to access.)

https://github.com/BurntSushi/walkdir

Reply:
do we want to ignore bad files?

Reply:
you should at least log what file is bad